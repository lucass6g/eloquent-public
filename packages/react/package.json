{
  "name": "@eloquent/react",
  "version": "1.0.0",
  "description": "",
  "main": "./dist/index.js",
  "files": [
    "dist"
  ],
  "types": "./dist/index.d.ts",
  "exports": {
    ".": {
      "import": "./dist/index.mjs",
      "require": "./dist/index.js",
      "types": "./dist/index.d.ts"
    }
  },
  "scripts": {
    "build": "tsup",
    "clean": "rm -rf dist && rm -rf node_modules"
  },
  "keywords": [],
  "author": "",
  "license": "MIT",
  "devDependencies": {
    "tsup": "^8.1.0",
    "@types/react": "^18.3.3",
    "@types/react-dom": "^18.3.0",
    "react": "^18.3.1",
    "typescript": "^5.4.5"
  }
}
