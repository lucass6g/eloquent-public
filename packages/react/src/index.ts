export {Switch, Case, Default} from './components/conditional-switch'