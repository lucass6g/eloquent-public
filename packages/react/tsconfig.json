{
  "extends": "../../packages/config/base.tsconfig.json",
  "include": ["src"],
  "exclude": ["dist", "build", "node_modules"],
  "compilerOptions": {
    "forceConsistentCasingInFileNames": false,
    "rootDir": "src",
    "baseUrl": ".",
    "paths": {
      "@/*": ["./src/*"]
    }
  }
}
