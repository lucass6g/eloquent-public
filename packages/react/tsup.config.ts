import { defineConfig } from 'tsup';

export default defineConfig({
	name: "@eloquent/react",
	entry: ["src/index.ts"],
	outDir: "dist",
	target: "es2022",
	format: ["cjs", "esm"],
	splitting: false,
	sourcemap: true,
	clean: true,
	dts: true,
	external: ["react", 'react-dom', 'react/jsx-runtime'],
	cjsInterop: true,
	treeshake: true,
})