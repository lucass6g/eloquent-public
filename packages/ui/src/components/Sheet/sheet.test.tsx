import { fireEvent, render, RenderResult, screen } from '@testing-library/react';
import { it, describe, expect, vi, beforeEach, afterEach } from "vitest"
import { Sheet } from '.';

const SheetTest = (props: React.ComponentProps<typeof Sheet.Root>) => (
    <Sheet.Root>
        <Sheet.Trigger>Abrir</Sheet.Trigger>
        <Sheet.Content {...props}>
            <Sheet.Header>
                <Sheet.Title>Você está certo disso?</Sheet.Title>
                <Sheet.Description>
                    Essa ação não pode ser desfeita. Isso excluirá permanentemente sua conta
                    e removerá seus dados dos nossos servidores.
                </Sheet.Description>
            </Sheet.Header>
        </Sheet.Content>
    </Sheet.Root>
);

describe('given a default Sheet', () => {
    let rendered: RenderResult;
    let trigger: HTMLElement;
    let closeButton: HTMLElement;
    let consoleWarnMock: ReturnType<typeof vi.spyOn>;
    let consoleWarnMockFunction: ReturnType<typeof vi.fn>;
    let consoleErrorMock: ReturnType<typeof vi.spyOn>;
    let consoleErrorMockFunction: ReturnType<typeof vi.fn>;

    beforeEach(() => {
        consoleWarnMockFunction = vi.fn();
        consoleWarnMock = vi.spyOn(console, 'warn').mockImplementation(consoleWarnMockFunction);
        consoleErrorMockFunction = vi.fn();
        consoleErrorMock = vi.spyOn(console, 'error').mockImplementation(consoleErrorMockFunction);

        rendered = render(<SheetTest />);
        trigger = rendered.getByText('Abrir');
    });

    afterEach(() => {
        consoleWarnMock.mockRestore();
        consoleWarnMockFunction.mockClear();
        consoleErrorMock.mockRestore();
        consoleErrorMockFunction.mockClear();
    });


    describe('after clicking the trigger', () => {
        beforeEach(() => {
            fireEvent.click(trigger);
            closeButton = rendered.getByText('Close');
        });

        it('should render content correctly', () => {
            const component = render(<SheetTest />)
            expect(screen.getByText('Essa ação não pode ser desfeita. Isso excluirá permanentemente sua conta e removerá seus dados dos nossos servidores.')).toBeInTheDocument();
            expect(component.baseElement).toMatchSnapshot();
        });


        it('should open the content', () => {
            expect(closeButton).toBeVisible();
        });

        describe('when pressing escape', () => {
            beforeEach(() => {
                fireEvent.keyDown(document.activeElement!, { key: 'Escape' });
            });

            it('should close the content', () => {
                expect(closeButton).not.toBeInTheDocument();
            });
        });
    });
});