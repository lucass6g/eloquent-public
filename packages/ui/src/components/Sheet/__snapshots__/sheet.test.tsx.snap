// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`given a default Sheet > after clicking the trigger > should render content correctly 1`] = `
<body
  data-scroll-locked="1"
  style="pointer-events: none;"
>
  <span
    aria-hidden="true"
    data-aria-hidden="true"
    data-radix-focus-guard=""
    style="outline: none; opacity: 0; position: fixed; pointer-events: none;"
    tabindex="0"
  />
  <div
    aria-hidden="true"
    data-aria-hidden="true"
  >
    <button
      aria-controls="radix-:r0:"
      aria-expanded="true"
      aria-haspopup="dialog"
      data-state="open"
      type="button"
    >
      Abrir
    </button>
  </div>
  <div
    aria-hidden="true"
    class="fixed inset-0 z-50 bg-black/80 data-[state=open]:animate-in data-[state=closed]:animate-out data-[state=closed]:fade-out-0 data-[state=open]:fade-in-0"
    data-aria-hidden="true"
    data-state="open"
    style="pointer-events: auto;"
  />
  <div
    aria-describedby="radix-:r2:"
    aria-labelledby="radix-:r1:"
    class="fixed z-50 gap-4 bg-background p-6 shadow-lg transition ease-in-out data-[state=open]:animate-in data-[state=closed]:animate-out data-[state=closed]:duration-300 data-[state=open]:duration-500 inset-y-0 right-0 h-full w-3/4 border-l data-[state=closed]:slide-out-to-right data-[state=open]:slide-in-from-right sm:max-w-sm"
    data-state="open"
    id="radix-:r0:"
    role="dialog"
    style="pointer-events: auto;"
    tabindex="-1"
  >
    <div
      class="flex flex-col space-y-2 text-center sm:text-left"
    >
      <h2
        class="text-lg font-semibold text-foreground"
        id="radix-:r1:"
      >
        Você está certo disso?
      </h2>
      <p
        class="text-sm text-muted-foreground"
        id="radix-:r2:"
      >
        Essa ação não pode ser desfeita. Isso excluirá permanentemente sua conta e removerá seus dados dos nossos servidores.
      </p>
    </div>
    <button
      class="absolute right-4 top-4 rounded-sm opacity-70 ring-offset-background transition-opacity hover:opacity-100 focus:outline-none focus:ring-2 focus:ring-ring focus:ring-offset-2 disabled:pointer-events-none data-[state=open]:bg-secondary [&>svg]:h-4 [&>svg]:w-4 [&>span]:sr-only"
      type="button"
    >
      <svg
        fill="none"
        height="15"
        viewBox="0 0 15 15"
        width="15"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clip-rule="evenodd"
          d="M11.7816 4.03157C12.0062 3.80702 12.0062 3.44295 11.7816 3.2184C11.5571 2.99385 11.193 2.99385 10.9685 3.2184L7.50005 6.68682L4.03164 3.2184C3.80708 2.99385 3.44301 2.99385 3.21846 3.2184C2.99391 3.44295 2.99391 3.80702 3.21846 4.03157L6.68688 7.49999L3.21846 10.9684C2.99391 11.193 2.99391 11.557 3.21846 11.7816C3.44301 12.0061 3.80708 12.0061 4.03164 11.7816L7.50005 8.31316L10.9685 11.7816C11.193 12.0061 11.5571 12.0061 11.7816 11.7816C12.0062 11.557 12.0062 11.193 11.7816 10.9684L8.31322 7.49999L11.7816 4.03157Z"
          fill="currentColor"
          fill-rule="evenodd"
        />
      </svg>
      <span>
        Close
      </span>
    </button>
  </div>
  <span
    aria-hidden="true"
    data-aria-hidden="true"
    data-radix-focus-guard=""
    style="outline: none; opacity: 0; position: fixed; pointer-events: none;"
    tabindex="0"
  />
  <div>
    <button
      aria-controls="radix-:r3:"
      aria-expanded="false"
      aria-haspopup="dialog"
      data-state="closed"
      type="button"
    >
      Abrir
    </button>
  </div>
</body>
`;
